pkg/commands/convert/run.go

```diff
@@ -0,0 +1,43 @@
+package convert
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/trivy/pkg/commands/operation"
+	"github.com/aquasecurity/trivy/pkg/flag"
+	"github.com/aquasecurity/trivy/pkg/log"
+	"github.com/aquasecurity/trivy/pkg/report"
+	"github.com/aquasecurity/trivy/pkg/result"
+	"github.com/aquasecurity/trivy/pkg/types"
+)
+
+func Run(ctx context.Context, opts flag.Options) (err error) {
+	f, err := os.Open(opts.Target)
+	if err != nil {
+		return xerrors.Errorf("file open error: %w", err)
+	}
+	defer f.Close()
+
+	var r types.Report
+	if err = json.NewDecoder(f).Decode(&r); err != nil {
+		return xerrors.Errorf("json decode error: %w", err)
+	}
+
+	if err = result.Filter(ctx, r, opts.FilterOpts()); err != nil {
```

Comment:
we need exclude some result from report.
e.g. secrets/misconfigs for spdx report:
```
➜  ls -hl 
-rw-rw-r-- 1 dmitriy dmitriy  891 авг  3  2019 server.key

➜  ./trivy -d fs -f json -o report.json .    
...
2023-05-24T10:19:50.094+0600	INFO	Number of language-specific files: 0
2023-05-24T10:19:50.094+0600	DEBUG	Secret file: server.key
➜   ./trivy -d convert -f spdx-json report.json   
2023-05-24T10:20:09.315+0600	DEBUG	["cyclonedx" "spdx" "spdx-json" "github"] automatically enables '--list-all-pkgs'.
2023-05-24T10:20:09.315+0600	DEBUG	Severities: ["UNKNOWN" "LOW" "MEDIUM" "HIGH" "CRITICAL"]
2023-05-24T10:20:09.315+0600	INFO	"--format spdx" and "--format spdx-json" disable security scanning
2023-05-24T10:20:09.315+0600	DEBUG	Writing report to output...
2023-05-24T10:20:09.316+0600	FATAL	unable to write results:
    github.com/aquasecurity/trivy/pkg/commands/convert.Run
        /home/dmitriy/work/aquasecurity/trivy/pkg/commands/convert/run.go:36
  - failed to write results:
    github.com/aquasecurity/trivy/pkg/report.Write
        /home/dmitriy/work/aquasecurity/trivy/pkg/report/writer.go:137
  - failed to save spdx json:
    github.com/aquasecurity/trivy/pkg/report/spdx.Writer.Write
        /home/dmitriy/work/aquasecurity/trivy/pkg/report/spdx/spdx.go:39
  - json: error calling MarshalJSON for type common.DocElementID: failed to marshal empty DocElementID
```

Reply:
How did you reproduce it? It works in my env.

Reply:
Oh, `-f spdx` works, but `-f spdx-json` doesn't. I'll look into it. Thanks.

Reply:
```
➜  4452 cat Dockerfile
FROM alpine

➜  4452 ./trivy -d config -f json -o report.json ./Dockerfile
2023-05-24T13:11:13.664+0600	DEBUG	Severities: ["UNKNOWN" "LOW" "MEDIUM" "HIGH" "CRITICAL"]
2023-05-24T13:11:13.667+0600	DEBUG	cache dir:  /home/dmitriy/.cache/trivy
2023-05-24T13:11:13.667+0600	INFO	Misconfiguration scanning is enabled
2023-05-24T13:11:13.668+0600	DEBUG	Policies successfully loaded from disk
2023-05-24T13:11:13.689+0600	DEBUG	Walk the file tree rooted at 'Dockerfile' in parallel
2023-05-24T13:11:13.689+0600	DEBUG	Scanning Dockerfile files for misconfigurations...
2023-05-24T13:11:14.194+0600	DEBUG	OS is not detected.
2023-05-24T13:11:14.194+0600	INFO	Detected config files: 1
2023-05-24T13:11:14.194+0600	DEBUG	Scanned config file: Dockerfile
➜  4452 ./trivy convert -f spdx-json report.json
2023-05-24T13:11:38.856+0600	INFO	"--format spdx" and "--format spdx-json" disable security scanning
2023-05-24T13:11:38.857+0600	FATAL	unable to write results: failed to write results: failed to save spdx json: json: error calling MarshalJSON for type common.DocElementID: failed to marshal empty DocElementID

```

Reply:
Fixed in https://github.com/aquasecurity/trivy/pull/4452/commits/73fa6bdcf7c940954039d2b3f662021d79e485fd